### PR TITLE
Pulsar Scaler: fix `msgBacklogThreshold` field being named wrongly as `msgBacklog`

### LINKED ISSUE #4681: Keda Pulsar integration doc says `msgBacklogThreshold` is the parameter but code is looking for `msgBacklog`
### Report

Compare https://keda.sh/docs/2.10/scalers/pulsar/
mentioning
`    msgBacklogThreshold: '5'`
to 
https://github.com/kedacore/keda/blob/2234a6f30df6b9fa83a1b88f1aa609ebee846983/pkg/scalers/pulsar_scaler.go#L43

### Expected Behavior

Code matches documentation 

### Actual Behavior

Code does not match documentation - overriding `msgBacklogThreshold` has no impact

### Steps to Reproduce the Problem

1. Compare Docs and code
2.
3.

### Logs from KEDA operator

```
N/A
```

### KEDA Version

2.10.1

### Kubernetes Version

< 1.23

### Platform

Amazon Web Services

### Scaler Details

Pulsar

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dttung2905**
Hi team,
This PR aims to ix `msgBacklogThreshold` field being named wrongly as `msgBacklog` . I pass down the logger into the method  `parsePulsarMetadata` as well

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4681

### CODE REVIEW

File: pkg/scalers/pulsar_scaler.go [hunk 1]
```diff
@@ -178,13 +179,24 @@ func parsePulsarMetadata(config *ScalerConfig) (pulsarMetadata, error) {
 
 	meta.msgBacklogThreshold = defaultMsgBacklogThreshold
 
+	// FIXME: msgBacklog support DEPRECATED to be removed in v2.13
+	fmt.Println(config.TriggerMetadata)
 	if val, ok := config.TriggerMetadata[msgBacklogMetricName]; ok {
+		logger.V(1).Info("\"msgBacklog\" is deprecated and will be removed in v2.13, please use \"msgBacklogThreshold\" instead")
```
Comment (JorTurFer) [inline pkg/scalers/pulsar_scaler.go:185]: v2.11 has already been released and this change will be included in v2.12, so the removal will be in 2.14
```suggestion
		logger.V(1).Info("\"msgBacklog\" is deprecated and will be removed in v2.14, please use \"msgBacklogThreshold\" instead")
```

File: pkg/scalers/pulsar_scaler.go [hunk 2]
```diff
@@ -178,13 +179,24 @@ func parsePulsarMetadata(config *ScalerConfig) (pulsarMetadata, error) {
 
 	meta.msgBacklogThreshold = defaultMsgBacklogThreshold
 
+	// FIXME: msgBacklog support DEPRECATED to be removed in v2.13
```
Comment (JorTurFer) [inline pkg/scalers/pulsar_scaler.go:182]:
```suggestion
	// FIXME: msgBacklog support DEPRECATED to be removed in v2.14
```

**Comment by JorTurFer** [comment]
Thanks for the PR, ÇI have left some small nits

File: pkg/scalers/pulsar_scaler.go (hunk 1 above)
Comment (dttung2905) [reply pkg/scalers/pulsar_scaler.go:185]: Thank @JorTurFer for the review. I will apply the same fix on the test case too!

**Review by JorTurFer — COMMENTED** [review]
LGTM!
Thanks for the fix ❤️  Could you update the changelog as well to reflect this?

**Comment by dttung2905** [comment]
Ah ha, no wonder I have a feeling that I was missing something :laughing:

**Review by JorTurFer — APPROVED** [review]
LGTM! ❤️

**Pull request merged**